normandy/recipes/signing.py

```diff
@@ -139,18 +144,8 @@ def verify_signature_pubkey(data, signature, pubkey):
             raise WrongSignatureSize("Base64 encoded signature was not a multiple of 4")
         else:
             raise
-    except AssertionError as e:
-        # The signature decoder has a clause like
-        #     assert len(signature) == 2*l, (len(signature), 2*l)
-        # If the AssertionError is consistent with that signature, translate it
-        # to a nicer error. Otherwise re-raise.
-        if (
-            len(e.args) == 1
-            and isinstance(e.args[0], tuple)
-            and len(e.args[0]) == 2
-            and isinstance(e.args[0][0], int)
-            and isinstance(e.args[0][1], int)
-        ):
+    except ecdsa.util.MalformedSignature as e:
+        if ECDSA_WRONG_LENGTH_RE.match(e.args[0]):
```

Comment:
I'm not a big fan of doing regex matching against exception text bodies, that's pretty fragile.  Is there not some better way to distinguish which class of exception this is?

Reply:
This particular class of exception is raised in two places in the library. However, I believe this is the only way this class of exception can be raised by this function. We could skip the check and just assume it means `WrongSignatureSize`.

Reply:
That feels better to me.